### PR TITLE
Adding ledger tests

### PR DESCRIPTION
**Pull request by algonautshant**
Tests added to cover:
ledger.GetLastCatchpointLabel
ledger.ledger.ListAssets
ledger.ListApplications

### CODE REVIEW

**Review by tsachiherman — CHANGES_REQUESTED** [review]
Could you please rollback the go.mod ? I'm pretty sure it's not needed for this change.

File: ledger/ledger_test.go [hunk 1]
```diff
@@ -1078,3 +1078,137 @@ func TestLedgerReload(t *testing.T) {
...
+			id <= maxAsset {
+			assetCount++
+		}
+	}
+	require.Equal(t, assetCount, len(results))
+
+	// Test ListApplications
+	// Check the number of results limit
+	ledger.accts.creatables = randomCtbs
+	results, err = ledger.ListApplications(basics.AppIndex(maxApp), 2)
```
Comment (tsachiherman) [inline ledger/ledger_test.go:1201]: you should do something with the `results` variable here. Maybe test it's size like you did with the assets above ?
if that's not applied to this function, then just do `_, err = ledger.ListApplication .... `

**Review by tsachiherman — DISMISSED** [review]
Looks good, just one request regarding the `ledger.ListApplications(basics.AppIndex(maxApp), 2)` statement.

File: ledger/ledger_test.go [hunk 2]
```diff
@@ -1078,3 +1078,137 @@ func TestLedgerReload(t *testing.T) {
...
+	require.Equal(t, lastCatchpointLabel, ledger.GetLastCatchpointLabel())
+}
+
+// generate at least 3 asset and 3 app creatables, and return the ids
+// of the asset/app with at least 3 elements less or equal.
+func generateCreatables(numElementsPerSegement int) (
+	randomCtbs map[basics.CreatableIndex]modifiedCreatable,
+	assetID3,
+	appID3 basics.CreatableIndex,
+	err error) {
```
Comment (algorandskiy) [inline ledger/ledger_test.go:1109]: nit/style: hard to read ') {` on the next line might help:

```
func generateCreatables(numElementsPerSegement int) (
	randomCtbs map[basics.CreatableIndex]modifiedCreatable,
	assetID3,
	appID3 basics.CreatableIndex,
	err error
) {
	_, randomCtbs = randomCreatables(numElementsPerSegement)
...
}
```

**Review by algorandskiy — COMMENTED** [review]
Looks good, agree with @tsachiherman's notes

File: ledger/ledger_test.go (hunk 1 above)
Comment (algonautshant) [reply ledger/ledger_test.go:1201]: Yep. forgot the test the count here.

File: ledger/ledger_test.go (hunk 2 above)
Comment (algonautshant) [reply ledger/ledger_test.go:1109]: np

Comment (algonautshant) [reply ledger/ledger_test.go:1109]: will adding a new line help? 
otherwise, will need to add a ',' after error. Not nice.

**Comment by tsachiherman** [comment]
Could you please rebase the change, so the it won't include the `ledger/accountdb_test.go` and `goalAppAccountAddressTest.exp` changes ? looks good otherwise.

**Review by tsachiherman — APPROVED** [review]
looks great!

**Pull request merged**